src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+
+          const connectedAddress = options.connectedAddress;
+          const relayPublicKeyPath = options.relayKey;
+          const relayPublicKey = fs.readFileSync(relayPublicKeyPath).toString();
+
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress, relayPublicKey);
+        }
+
+        if (successful) {
+          pkLogger('peer successfully added to peer store', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong, peer was not added to peer store', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeUpdatePeerInfoCommand() {
+  return new commander.Command('update')
+    .description('update the peer info for a particular public key')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-c, --current-node', 'only list the peer information for the current node, useful for sharing')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-ch, --connected-host <connectedHost>', 'update the connected addresss host')
+    .option('-cp, --connected-port <connectedPort>', 'update the connected addresss port')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const currentNode: boolean = options.currentNode;
+
+        const publicKeyPath = options.publicKey;
+        // read in publicKey if it exists
+        let publicKey: string | undefined = undefined;
+        if (publicKeyPath) {
+          publicKey = fs.readFileSync(publicKeyPath).toString();
+        }
+
+        const relayPublicKeyPath = options.relayKey;
+        // read in relay publicKey if it exists
+        let relayPublicKey: string | undefined = undefined;
+        if (relayPublicKeyPath) {
+          relayPublicKey = fs.readFileSync(relayPublicKeyPath).toString();
+        }
+
+        const base64String = options?.base64?.replace('\r', '')?.replace('\n', '');
+        let successful: boolean;
+        if (base64String != undefined) {
+          // read in peer info string
+          const { publicKey, connectedAddress, relayPublicKey } = PeerInfo.parseB64(base64String);
+          successful = await client.updatePeer(
+            nodePath,
+            currentNode ? undefined : publicKey,
+            currentNode,
+            connectedAddress?.host,
+            connectedAddress?.port,
+            relayPublicKey,
+          );
+        } else {
+          successful = await client.updatePeer(
```

Comment:
If the client is a proxy representation of the agent. Then I think you are using the client as almost the RPC object too, and then calling all methods from there.

I think there's a better way do this.

Instead create models like in MVC. And then have the models be constructed with the agent object.

Sort of like database models that take the `db` object and sending commands to the `db` object.

Then you create an interface for the `Agent`. This allows you mock models.

The models are then collections of stateful operations.

This also allows you to be more cohesive and have interface composition. That is breaking up the BIG polykey interface into smaller interfaces that can be joined together.